pkg/controllers/management/eksupstreamrefresh/eks_upstream_refresh.go

```diff
@@ -182,10 +207,26 @@ func GetComparableUpstreamSpec(secretsCache wranglerv1.SecretCache, cluster *mgm
 		}
 
 		nodeGroupStates = append(nodeGroupStates, ng)
+		var nodeGroupMustBeDeleted string
+		if len(ng.Nodegroup.Health.Issues) != 0 {
+			var issueMessages []string
+			for _, issue := range ng.Nodegroup.Health.Issues {
+				issueMessages = append(issueMessages, aws.StringValue(issue.Message))
+				if !controller.NodeGroupIssueIsUpdatable(aws.StringValue(issue.Code)) {
+					nodeGroupMustBeDeleted = ": node group cannot be updated, must be deleted and recreated"
+				}
+			}
+			errs = append(errs, fmt.Sprintf("health error for node group [%s]: %s%s",
+				aws.StringValue(ng.Nodegroup.NodegroupName),
+				strings.Join(issueMessages, "; "),
+				nodeGroupMustBeDeleted,
```

Comment:
This will always append an empty string to the errs

Reply:
The last `nodeGroupMustBeDelete` is in the `fmt.Sprintf` call. The phrase `: node group cannot be updated, must be deleted and recreated` is being added to the error string if the check `nodeGroupIssueIsUpdatable` fails.

Reply:
Yes, bad phrasing on my part. If `nodeGroupMustBeDeleted` is not set on line 221, then this will still append an empty string. Does that end up in ux/logs somehow ?

Reply:
This error string does eventually show up in the UI.

If `nodeGroupMustBeDeleted` is empty, then it will be put into the last format string verb and will be undetectable in the formatted string.